pymc3/sampling.py

```diff
@@ -419,6 +419,29 @@ def sample(
 
     """
     model = modelcontext(model)
+    if start is None:
+        start = model.test_point
+    else:
+        if isinstance(start, dict):
+            update_start_vals(start, model.test_point, model)
+        else:
+            for chain_start_vals in start:
+                update_start_vals(chain_start_vals, model.test_point, model)
+
+    start_points = [start] if isinstance(start, dict) else start
```

Comment:
If I remember correctly, the downstream code treats a `list` of `start` as "start points for each chain", which could explain your index error.